verified-broadcast/assets/broadcast.html

```diff
@@ -0,0 +1,187 @@
+            messages are sent.
+          </p>
+        </section>
+        <section>
+          <form id="broadcast">
+            <label for="message-body">Message</label>
+            <input type="text" id="message-body" />
+            <p>Send updates to:</p>
+            <div>
+              <input type="radio" id="all" name="tags" value="all" checked />
+              <label for="all">Everyone</label><br />
+            </div>
+            <div>
+              <input type="radio" id="option1" name="tags" value="option1" />
+              <label for="option1">Team Pie</label><br />
+            </div>
+            <div>
+              <input type="radio" id="option2" name="tags" value="option2" />
+              <label for="option2">Team Cake</label><br />
+            </div>
+            <input type="submit" value="Send message" />
+          </form>
+          <span id="status"></span>
+          <form id="otp" style="display: none">
+            <label for="code">Code</label>
+            <input type="text" id="code" />
+            <input type="submit" value="Confirm" />
+          </form>
+        </section>
+      </div>
+    </main>
+    <footer>
+      <span>We can't wait to see what you build.</span>
+    </footer>
+  </body>
+  <script>
+    function logError(error) {
+      console.log(error);
+      document.getElementById("status").textContent = error;
+    }
+
+    function sendVerificationToken(event) {
+      event.preventDefault();
+
+      fetch("./start-verify")
+        .then((response) => response.json())
+        .then((json) => {
+          if (json.success) {
+            document.getElementById("intro").style.display = "none";
+            document.getElementById("broadcast").style.display = "none";
+
+            document.getElementById("status").textContent =
+              "Successfully sent token. Enter token sent to Admin to broadcast message.";
+
+            // show OTP form
+            document.getElementById("otp").style.display = "flex";
+          } else {
+            logError(json.error);
+          }
+        })
+        .catch((error) => {
+          logError(error);
+        });
+    }
+
+    function verify(event) {
+      event.preventDefault();
+
+      const body = document.getElementById("message-body").value;
+      const code = document.getElementById("code").value;
+
+      const data = new URLSearchParams();
+      data.append("body", body);
+      data.append("code", code);
+
+      var tag = document.querySelector('input[name="tags"]:checked').value;
+      data.append("tag", tag);
+
+      fetch("./broadcast", {
+        method: "POST",
+        body: data,
+      })
+        .then((response) => response.json())
+        .then((json) => {
+          if (json.success) {
+            document.getElementById("otp").style.display = "none";
+            document.getElementById("status").textContent =
+              "Successfully sent message to subscribers.";
```

Comment:
It might be nice to add a button that allows you to send another broadcast (return the original form to the page) here. Not a blocker, but an idea for the experience of using it.

Reply:
Good idea - added